### DISCUSSION TITLE
0

### DISCUSSION THREAD

**Discussion by dzenthai** [rubenlagus/TelegramBots#1440] [General]

**Comment by Chase22** [comment]
The split was made to have a separation of concerns between interacting with the API and processing updates. This separation should be reflected in your code. Components that require you to talk to the API need an API client.

There's nothing stopping you from having multiple API clients in your program, but it's much better to use patterns like dependency injection or singletons to manage the instance. 

If you have concerns about the API of the library other than "they ruined it" feel free to bring them up and we can improve the API
